dockstore-integration-testing/src/test/java/io/dockstore/webservice/ServiceIT.java

```diff
@@ -148,6 +153,104 @@ public void testGeneralDefaultPathMechanism() {
         final io.swagger.client.model.Workflow workflow = client.getWorkflow(invoke.getServiceID(), "");
     }
 
+    /**
+     * This tests endpoints that will be triggered by GitHub App webhooks.
+     * A service is created and a version is added for a release 1.0
+     */
+    @Test
+    public void testGitHubAppEndpoints() throws Exception {
+        CommonTestUtilities.cleanStatePrivate2(SUPPORT, false);
+        final ApiClient webClient = getWebClient("admin@admin.com");
+        WorkflowsApi client = new WorkflowsApi(webClient);
+
+        String serviceRepo = "DockstoreTestUser2/test-service";
+        String installationId = "1179416";
+
+        // Add service
+        io.swagger.client.model.Workflow service = client.addService(serviceRepo, "admin@admin.com", installationId);
+        assertNotNull(service);
+
+        // Add version with another username
+        service = client.upsertServiceVersion(serviceRepo, "DockstoreTestUser2", "1.0", installationId);
+
+        assertNotNull(service);
+        assertEquals("Should have a new version", 1, service.getWorkflowVersions().size());
+        assertEquals("Should have 3 source files", 3, service.getWorkflowVersions().get(0).getSourceFiles().size());
+        assertEquals("Should have 2 users", 2, service.getUsers().size());
+    }
+
+    /**
+     * Ensures that you cannot create a service if the given user is not on Dockstore
+     */
+    @Test
+    public void createServiceNoUser() throws Exception {
+        CommonTestUtilities.cleanStatePrivate2(SUPPORT, false);
+        final ApiClient webClient = getWebClient("admin@admin.com");
+        WorkflowsApi client = new WorkflowsApi(webClient);
+
+        String serviceRepo = "DockstoreTestUser2/test-service";
+        String installationId = "1179416";
+
+        // Add service
+        io.swagger.client.model.Workflow service = null;
+        try {
+            service = client.addService(serviceRepo, "iamnotarealuser", installationId);
+        } catch (ApiException ex) {
+
```

Comment:
Not a big deal, but this makes me nervous, this could fail in a different way and still pass the test. 
Sometimes I prefer to check the database instead.